interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/SubprocessDefinitionExtractor.scala

```diff
@@ -0,0 +1,75 @@
+package pl.touk.nussknacker.engine.definition
+
+import pl.touk.nussknacker.engine.api.component.{ParameterConfig, SingleComponentConfig}
+import pl.touk.nussknacker.engine.api.definition.Parameter
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, Unknown}
+import pl.touk.nussknacker.engine.api.{FragmentSpecificData, MetaData}
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.engine.canonicalgraph.canonicalnode.FlatNode
+import pl.touk.nussknacker.engine.definition.DefinitionExtractor.ObjectDefinition
+import pl.touk.nussknacker.engine.definition.SubprocessDefinitionExtractor.extractSubprocessParam
+import pl.touk.nussknacker.engine.definition.parameter.ParameterData
+import pl.touk.nussknacker.engine.definition.parameter.defaults.{DefaultValueDeterminerChain, DefaultValueDeterminerParameters}
+import pl.touk.nussknacker.engine.definition.parameter.editor.EditorExtractor
+import pl.touk.nussknacker.engine.definition.parameter.validator.{ValidatorExtractorParameters, ValidatorsExtractor}
+import pl.touk.nussknacker.engine.graph.node.SubprocessInputDefinition
+import pl.touk.nussknacker.engine.graph.node.SubprocessInputDefinition.SubprocessParameter
+
+
+case class SubprocessDetails(canonical: CanonicalProcess, category: String)
+
+
+class SubprocessDefinitionExtractor(category: String, subprocessesDetails: Set[SubprocessDetails], componentsConfig: Map[String, SingleComponentConfig], classLoader: ClassLoader) {
+
+  def extractBySubprocessId(subprocessId: String): List[Parameter] = {
+    extract.getOrElse(subprocessId, ObjectDefinition.withParams(List.empty)).parameters
```

Comment:
Why this `orElse` instead of returning None or sth? Please follow the fail fast approach, we are not in java script ;-)